tests/test_bases.py

```diff
@@ -42,3 +99,24 @@ def test_get_root_lengths(canola_h5):
     np.testing.assert_array_almost_equal(
         root_lengths, [20.129579, 62.782368, 80.268003, 34.925591, 3.89724]
     )
+
+
+def test_get_root_lengths_no_roots():
+    pts = np.array(
+        [
+            [
+                [np.nan, np.nan],
+                [np.nan, np.nan],
+            ],
+            [
+                [np.nan, np.nan],
+                [np.nan, np.nan],
+            ],
+        ]
+    )
+
+    assert pts.shape == (2, 2, 2)
+
+    root_lengths = get_root_lengths(pts)
+    assert root_lengths.shape == (2,)
+    np.testing.assert_array_almost_equal(root_lengths, [0, 0])
```

Comment:
Since the values are exact here, could use `np.testing.array_equal`

Also, it might be more appropriate to return NaNs in this case

Reply:
I changed `get_root_lengths` to return np.empty((len(pts),)) ** np.nan